go/oasis-node/cmd/config/migrate/migrate.go

```diff
@@ -0,0 +1,607 @@
+		mDebug := m(m(newCfg["common"])["debug"])
+
+		if rlimit, ok := m(debug)["rlimit"]; ok {
+			mDebug["rlimit"] = rlimit
+			delete(m(debug), "rlimit")
+		}
+
+		if allow_root, ok := m(debug)["allow_root"]; ok {
+			mDebug["allow_root"] = allow_root
+			delete(m(debug), "allow_root")
+		}
+	}
+
+	pprof, ok := oldCfg["pprof"]
+	if ok {
+		if bind, ok := m(pprof)["bind"]; ok {
+			logger.Info("pprof.bind is now pprof.bind_address")
+			mkSubMap(newCfg, "pprof")
+			m(newCfg["pprof"])["bind_address"] = bind
+			delete(m(pprof), "bind")
+		}
+	}
+
+	ias, ok := oldCfg["ias"]
+	if ok {
+		mkSubMap(newCfg, "ias")
+		mIAS := m(newCfg["ias"])
+
+		if proxy, ok := m(ias)["proxy"]; ok {
+			if address, ok := m(proxy)["address"]; ok {
+				logger.Info("ias.proxy.address is now ias.proxy_address")
+				mIAS["proxy_address"] = address
+				delete(m(proxy), "address")
+			}
+		}
+
+		if debug, ok := m(ias)["debug"]; ok {
+			if skip_verify, ok := m(debug)["skip_verify"]; ok {
+				logger.Info("ias.debug.skip_verify is now ias.debug_skip_verify")
+				mIAS["debug_skip_verify"] = skip_verify
+				delete(m(debug), "skip_verify")
+			}
+		}
+	}
+
+	genesis, ok := oldCfg["genesis"]
+	if ok {
+		mkSubMap(newCfg, "genesis")
+		mGen := m(newCfg["genesis"])
+
+		if file, ok := m(genesis)["file"]; ok {
+			mGen["file"] = file
+			delete(m(genesis), "file")
+		}
+	}
+
+	// If we deleted any keys, make sure we don't retain empty structures.
+	pruneEmptyMaps(oldCfg)
+	// Also prune new config.
+	pruneEmptyMaps(newCfg)
+
+	// Check for options that are only available on the command-line.
+	if _, ok = oldCfg["debug"]; ok {
+		logger.Warn("note that some debug.* options are from now on only available on the command-line")
+	}
+	if _, ok = oldCfg["grpc"]; ok {
+		logger.Warn("note that grpc.* options are from now on only available on the command-line")
+	}
+	if _, ok = oldCfg["metrics"]; ok {
+		logger.Warn("note that metrics.* options are from now on only available on the command-line")
+	}
+
+	// Print sections remaining in map (if any).
+	if len(oldCfg) > 0 {
+		remaining, grr := yaml.Marshal(&oldCfg)
+		if grr != nil {
+			logger.Error("failed to marshal remaining sections from input file", "err", grr)
+			os.Exit(1)
+		}
+
+		fmt.Printf("Ignored unknown sections from input file:\n%s\n", remaining)
+	}
+
+	// Save new config to file.
+	newCfgRaw, err := yaml.Marshal(&newCfg)
+	if err != nil {
+		logger.Error("failed to marshal converted config file", "err", err)
+		os.Exit(1)
+	}
+	if err = os.WriteFile(cfgOutFileName, newCfgRaw, 0o600); err != nil {
+		logger.Error("failed to write converted config file", "file_name", cfgOutFileName, "err", err)
+		os.Exit(1)
+	}
+
+	// Validate new config.
+	newCfgStruct := config.DefaultConfig()
+	err = yaml.Unmarshal(newCfgRaw, &newCfgStruct)
+	if err != nil {
+		logger.Error("failed to parse config file after migration (this might be normal if you're using environment variable substitutions in your original config file)", "err", err)
```

Comment:
This error message is a bit confusing for me, as I expect the problem is in the original file and not in the converted one. I hope the users won't have the same problem. 

Options:
- Add few extra lines, one printing that we are doing a migration, one that migration was completed, file saved, another that we are validating the migrated file.
- Or split migration/validation, e.g. create a new command and print the command for validation at the end. Maybe even add a warning that the file is not validated. New command could be useful latter to verify already migrated config files.
- Keep as it is.

Reply:
Config validation is done by default now when then node starts up and loads the config file, so I don't think a separate command is needed for that.

I will add a few more info logs to let the user know what's happening.